src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1126,17 +1158,21 @@ public UpdateTransaction newUpdateTransaction(PartitionUpdate update, WriteConte
     {
         if (!hasIndexes())
             return UpdateTransaction.NO_OP;
-
-        Index.Indexer[] indexers = indexes.values().stream()
-                                          .map(i -> i.indexerFor(update.partitionKey(),
-                                                                 update.columns(),
-                                                                 nowInSec,
-                                                                 ctx,
-                                                                 IndexTransaction.Type.UPDATE))
-                                          .filter(Objects::nonNull)
-                                          .toArray(Index.Indexer[]::new);
-
-        return indexers.length == 0 ? UpdateTransaction.NO_OP : new WriteTimeTransaction(indexers);
+        
+        ArrayList<Index.Indexer> idxrs = new ArrayList<>();
+        for (Index i : indexes.values())
+        {
+            Index.Indexer idxr = i.indexerFor(update.partitionKey(), update.columns(), nowInSec, ctx, IndexTransaction.Type.UPDATE);
+            if (idxr != null && isIndexWritable(i))
```

Comment:
We could check if the index is writable before creating the indexer:
```suggestion
            if (!isIndexWritable(i))
                continue;

            Index.Indexer idxr = i.indexerFor(update.partitionKey(), update.columns(), nowInSec, ctx, IndexTransaction.Type.UPDATE);
            if (idxr != null)
```
The same applies to the other usages of `isIndexWritable` in `IndexGCTransaction#commit()` and `CleanupGCTransaction#commit()`. Alternatively, we could transform the set `SIM#writableIndexes` into a `Map<String, Index>`, and iterate its values this instead of `SIM#indexes` values in the three places. That way we wouldn't even need the `isIndexWritable` method.

Reply:
Sgtm. That was my first implementation I can't remember for the sake of me why I changed it to a set.